presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -806,6 +806,13 @@ protected Scope visitJoin(Join node, Scope scope)
 
                 addCoercionForJoinCriteria(node, leftExpression, rightExpression);
                 expressions.add(new ComparisonExpression(EQUAL, leftExpression, rightExpression));
+
+                if (node.getType() == Join.Type.LEFT) {
+                    output = createScope(node, scope, left.getRelationType().joinWith(right.hide(rightExpression).getRelationType()));
```

Comment:
I need to re-read that part of the spec, but if I recall correctly, then qualified columns become unavailable in the scope and the join columns (there could be multiple), are equivalent to coalesce(l.x, r.x)